com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl

```diff
@@ -357,6 +363,12 @@ void EvaluateAdaptiveProbeVolume(in float3 posWS, in float3 normalWS, in float3
         bakeDiffuseLighting += apvSample.L0;
         backBakeDiffuseLighting += apvSample.L0;
         lightingInReflDir += apvSample.L0;
+
+        if (_Weight < 1.f)
```

Comment:
there is no need for this test here. don't think we save anything with it

Reply:
the train of thought was that since weight is likely to be 1 99% of the time we could branch if away cheaply based on the uniform here. (I'll admit I haven't actually checked the asm for this assumption yet)

Reply:
The branch is static so I guess we do skip the ALUs but I don't think we skip the loads caused by EvaluateAmbientProbe so we'd be adding potential cost all the time for a fringe case here.
Maybe it's ok since we do call EvaluateAmbientProbe in another branch anyway. Might need to look a bit more at the generated code to make sure of this.

Reply:
@JulienIgnace-Unity why would the loads not be skipped by the branch? at least looking at the assembly it looks like they all get gated behind branches.